sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessor.java

```diff
@@ -229,14 +236,48 @@ private void receiveEvents(PartitionOwnership partitionOwnership) {
         PartitionProcessor partitionProcessor = this.partitionProcessorFactory
             .createPartitionProcessor(partitionContext, checkpointManager);
         partitionProcessor.initialize().subscribe();
-
+        final AtomicReference<Context> processSpanContext = new AtomicReference<>(Context.NONE);
         consumer.receive().subscribeOn(Schedulers.newElastic("PartitionPump"))
-            .subscribe(eventData -> partitionProcessor.processEvent(eventData).subscribe(unused -> {
-            }, partitionProcessor::processError),
+            .subscribe(eventData -> {
+                startScopedTracingSpan(eventData, processSpanContext);
+                partitionProcessor.processEvent(eventData).subscribe(unused -> {
+                }, partitionProcessor::processError);
+                endScopedTracingSpan(processSpanContext);
+            },
                 partitionProcessor::processError,
                 // Currently, there is no way to distinguish if the receiver was closed because
                 // another receiver with higher/same owner level(epoch) connected or because
                 // this event processor explicitly called close on this consumer.
                 () -> partitionProcessor.close(CloseReason.LOST_PARTITION_OWNERSHIP));
     }
+
+    /*
+     * Starts a new process tracing span and attached context the EventData object for users.
+     */
+    private void startScopedTracingSpan(EventData eventData, AtomicReference<Context> processSpanContext) {
```

Comment:
this method doesn't seem like general-purpose one and only creates process spans, maybe is should be called like that?